### PR TITLE
oneapi/dpcpp 2021.2.0 docker changes.

### PR DESCRIPTION
**Pull request by mdavis36**
## Summary
Changes for building oneapi image with the latest dpcpp. 

This PR:
- Changes setvar script source to the new directory of the oneapi install location. 
- Uses the gcc-8 axom image as the base container to avoid compilation errors when compiling with dpcpp later on.

### CODE REVIEW

**Review by davidbeckingsale — APPROVED** [review]
Looks good, thanks Mike. For the benefit of other reviewers - these changes are required to support current version of OneAPI. The path changed, and base is changed to gcc-8 because SYCL/dpcpp are using C++17

**Review by rhornung67 — APPROVED** [review]
Thanks @mdavis36 Did you try this with the SYCL PR branch?

File: ubuntu-16/oneapi/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM axom/compilers:ubuntu-base
+FROM axom/compilers:gcc-8
```
Comment (white238) [inline ubuntu-16/oneapi/Dockerfile:1]: This brings up an issue that we lost information about which ubuntu this compiler tag is using.  For a follow up PR though.

**Comment by mdavis36** [comment]
> Thanks @mdavis36 Did you try this with the SYCL PR branch?

Yep, I have everything working locally using this container so once its built and pushed to dockerhub I can update RAJAs azure CI. Also can someone else merge this branch when ready, I don't have authorization.

**Comment by rhornung67** [comment]
@davidbeckingsale should we merge this?

**Comment by davidbeckingsale** [comment]
> @davidbeckingsale should we merge this?

Yup, this can be merged. I still need to build and push the image up to dockerhub, but can do that separately

**Pull request merged**